lib/srv/server/azure_watcher_test.go

```diff
@@ -99,7 +99,7 @@ func TestAzureWatcher(t *testing.T) {
 				Regions:        []string{"location1", "location2"},
 				ResourceTags:   types.Labels{"*": []string{"*"}},
 			},
-			wantVMs: []string{"vm1", "vm2"},
+			wantVMs: []string{"vm1", "vm2", "vm5"},
```

Comment:
I think it worked some of the time before because it polls and just writes instances to the channel as it sees them, and sometimes the polling saw 1 and 2 before 5 was written to the channel so the test passed.